lib/imports/api/authority-registry/server/authority-registry.js

```diff
@@ -0,0 +1,319 @@
+  * @returns {{}} authority
+  */
+  getAuthorityInfoByDocId ({ docId }) {
+    return authorityRegistryColl.findOne({ docIds: docId });
+  }
+
+  /**
+  *  Returns a single authority from the registry database, found via a groupId that lives in an authority.
+  * @param {Object} params
+  * @param {String} params.groupId        groupId that the wanted authority is hosting
+  * @returns {{}} authority
+  */
+  getAuthorityInfoByGroupId ({ groupId }) {
+    return authorityRegistryColl.findOne({ groupIds: groupId });
+  }
+  /**
+  *  Returns if the specified server is registered or not.
+  * @param {Object} params
+  * @param {String} params.ip     ip of the server to remove from the registry
+  * @param {Number} params.port   port of the server to remove from the registry
+  * @returns {Boolean} if the server is registered
+  */
+  isRegistered ({ ip, port }) {
+    check(ip, String);
+    check(port, Number);
+    return authorityRegistryColl.find({ ip, port }).count() !== 0;
+  }
+
+  /**
+  *  Adds a doc to an authority in the registry.
+  * @param {Object} params
+  * @param {String} params.docId      unique id for this doc
+  * @param {String} [params.groupId]  optional, a groupId of this doc to associate it with other docs
+  */
+  addDocToAuthority ({ docId, groupId = null }) {
+    addDocToAuthority.call({ docId, groupId, fullUrl: this.fullUrl });
+    console.log(`ProseMeteor AuthorityRegistry: Added doc ${docId} to authority ${this.fullUrl}. Updated registry: ${JSON.stringify(this.fullRegistry(), null, 2)}`);
+  }
+  /**
+  *  Removes a doc from an authority in the registry. If this doc is in a groupId and no more of that group are hosted
+  *  on this server, the groupId is also removed from this authority.
+  * @param {Object} params
+  * @param {String} params.docId      unique id for this doc
+  * @param {String} [params.groupId]  optional, a groupId of this doc to associate it with other docs
+  */
+  removeDocFromAuthority ({ docId, groupId = null }) {
+    // first remove this docId
+
+    if (groupId) {
+      // check each remaining docId. if any have the same groupId, keep this docs groupId. otherwise remove it from groupIds[]
+    }
+  }
+  /**
+  *  Chooses a list of eligible authority servers to host a document, preferring the server with the least documents.
+  * @returns {Array}  chosenAuthorities     array of chosen authority servers
+  */
+  getEligibleAuthoritiesToHostDoc () {
+    // otherwise get 1-5 servers hosting the lowest number of docs
+    const fullRegistry = this.fullRegistry();
+    const fullRegistrySortedByDocIdsCount = fullRegistry
+      .sort((authorityA, authorityB) => {
+        return authorityA.docIds.length > authorityB.docIds.lenth;
+      })
+      .slice(0, 5);
+    console.log(`ProseMeteor AuthorityRegistry: Selecting eligible authority servers with least documents: "${JSON.stringify(fullRegistrySortedByDocIdsCount, null, 2)}"`);
+
```

Comment:
It would be interesting to figure out where bottle neck would be with real users. Ideally what ever we go with should be very easy to scale at a constant rate of node per # of docs open.